### ISSUE TITLE
nunjucks included template with 11ty/image shortcode is not rendered

### ISSUE THREAD

**Issue by btrem** [11ty/eleventy#2589]
**Describe the bug**
If I have a responsive picture shortcode in an included nunjucks template, the template is not rendered. (Not just the shortcode--none of the code is rendered.)

**Steps To Reproduce**

1. Add [eleventy/image](https://www.11ty.dev/docs/plugins/image/) to an 11ty/eleventy project.
2. Create shortcode to process images.
3. Create a nunjucks template that uses that shortcode.
4. Use nunjucks `include` to include that template in another template.

**Alternate Steps To Reproduce**

1. Clone [demonstration repo from GitHub](https://github.com/btrem/11ty-2589).
2. Run `npm install`.
3. Run `npm start`.

**Expected behavior**
Included nunjucks template is rendered.

**Actual behavior**
Nothing in the included template is rendered.

**Environment:**
- MacOS 12.6
- Eleventy Version 1.0.2
- Eleventy/Image 1.1.0

[Demonstration available on GitHub](https://github.com/btrem/11ty-2589).

**Comment by pdehaan** [comment]
I wonder if the Nunjucks `{% include %}` doesn't like the async-ness of `.addNunjucksAsyncShortcode()`.
I did a quick search of https://github.com/11ty/eleventy-img/ but didn't see anything obvious (a few mentions of loops or async macros).

**Comment by pdehaan** [comment]
Actually, looking at your code, I think it is just an async loop issue (ref: https://mozilla.github.io/nunjucks/templating.html#asynceach):

```diff
diff --git a/src/lists/list-snippet-picture.njk b/src/lists/list-snippet-picture.njk
index 8228631..4ad342c 100644
--- a/src/lists/list-snippet-picture.njk
+++ b/src/lists/list-snippet-picture.njk
@@ -13,9 +13,9 @@ title: List of Posts Using Included Snippet With Shortcode
                 There is no list below.
             </p>
 
-{% for post in collections.all %}
+{% asyncEach post in collections.all %}
 {% include "snippet-picture.njk" %}
-{% endfor %}
+{% endeach %}
 
             <p>
                 If I remove the shortcode from the included template,
```

### OUTPUT

```html
  <body>
    <h1>List of Posts Using Included Snippet With Shortcode</h1>

    <main>
      <p>
        This page lists all posts using a nunjucks <code>include</code>
        statement to insert a snippet. The snippet has a shortcode to add each
        post's featured image as defined in its yaml front matter. But the list
        does not work. There is no list below.
      </p>

      <article>
        <h2>
          <a href="/post1.html">Post 1 Dirt</a>
        </h2>

        <picture>
          <source
            srcset="/images/dirt-500w.webp 500w, /images/dirt-900w.webp 900w"
            sizes="90vw"
            type="image/webp"
          />
          <source
            srcset="/images/dirt-500w.jpeg 500w, /images/dirt-900w.jpeg 900w"
            sizes="90vw"
            type="image/jpeg"
          />
          <img
            src="/images/dirt-500w.jpeg"
            alt=""
            loading="lazy"
            decoding="async"
          />
        </picture>
        ...
```

**Comment by btrem** [comment]
I too wondered if async was the problem. I tried to turn my picture shortcode function into a synchronous function, but couldn't make it work.

I also consulted the nunjucks docs, but I was looking at `include`. It didn't occur to me to look for an async loop. (You don't know what you don't know, as they say.)

So well spotted. I'm at a loss as to why it works when the shortcode is directly in the template (cf. list.html in the demo). But asyncEach solves the problem.
